### ISSUE TITLE
3x Speed Regression on master

### ISSUE THREAD

**Issue by jpfr** [open62541/open62541#2034]
We have a 3x speed regression on master as per the readspeed unit test.

Any ideas which commit could be responsible for this?
I checked the nodestore. But the changes there are not responsible for the regression.

-----

The readspeed unit test on 0.3 (Release build):
```
jpfr@jpfr-ThinkPad-T530:~/software/open62541/build$ ./bin/tests/check_server_readspeed 
Running suite(s): Service Speed
[1601-01-01 00:00:00.000 (UTC+0000)] warn/securitypolicy        No PKI plugin set. Accepting all certificates
[1601-01-01 00:00:00.000 (UTC+0000)] info/session       Connection 0 | SecureChannel 0 | Session 00000001-0000-0000-0000-000000000000 | AddNodes: No TypeDefinition; Use the default TypeDefinition for the Variable/Object
duration was 0.746388 s
retval is Good
100%: Checks: 1, Failures: 0, Errors: 0
```

The readspeed unit test on master (Release build):

```
jpfr@jpfr-ThinkPad-T530:~/software/open62541/build$ ./bin/tests/check_server_readspeed 
Running suite(s): Service Speed
[1601-01-01 00:00:00.000 (UTC+0000)] warn/securitypolicy        No PKI plugin set. Accepting all certificates
[1601-01-01 00:00:00.000 (UTC+0000)] info/session       Connection 0 | SecureChannel 0 | Session g=00000001-0000-0000-0000-000000000000 | AddNodes: No TypeDefinition for ns=1;s=the.answer; Use the default TypeDefinition for the Variable/Object
duration was 2.314021 s
retval is Good
100%: Checks: 1, Failures: 0, Errors: 0
```

**Comment by Pro** [comment]
Thanks @crafti5.

I used the following script (`check_speed.sh`) with git bisect:

```sh
#!/bin/bash

# git bisect start
# This is master
# git bisect good 0d856f97c13bc5457459df25a2a98d59ceb9e1b4
# This is just some commit in the past where the readspeed was good
# git bisect bad a9c66b46007953a54694484222a1ce16beb08a45
# git bisect run ./check_speed.sh


cd $HOME/test/open62541
git submodule update --init --recursive
rm -rf build
mkdir build
cd build
cmake -DUA_NAMESPACE_ZERO=FULL \
    -DCMAKE_BUILD_TYPE=Debug -DUA_BUILD_EXAMPLES=ON -DUA_ENABLE_PUBSUB=ON -DUA_ENABLE_PUBSUB_DELTAFRAMES=ON -DUA_ENABLE_PUBSUB_INFORMATIONMODEL=ON -DUA_ENABLE_ENCRYPTION=ON -DUA_ENABLE_DISCOVERY=ON \
    -DUA_ENABLE_DISCOVERY_MULTICAST=ON -DUA_BUILD_UNIT_TESTS=ON -DUA_ENABLE_COVERAGE=OFF \
    -DUA_ENABLE_UNIT_TESTS_MEMCHECK=OFF ..
make check_server_readspeed -j
./bin/tests/check_server_readspeed > out.txt
duration=$(grep -E -o "was [0-9]\.[0-9]+" out.txt | awk '{print $2}')
# On my machine the bad readspeed takes more than 2.2 secs
dur_max=1.7
echo "Duration = $duration\n"
cd ..
if [ 1 -eq "$(echo "$duration > $dur_max" | bc)" ]; then
	echo "Bad"
        exit 1
else
	echo "Good"
        echo 0
fi
```

This results in:

```
02734c310dc4f29a45879eeae1e1b7f63d9c9c12 is the first bad commit
commit 02734c310dc4f29a45879eeae1e1b7f63d9c9c12
Author: Julius Pfrommer <julius.pfrommer@web.de>
Date:   Fri Aug 3 04:27:35 2018 +0200

    Session: Print complete NodeId for the SessionToken

:040000 040000 bc49d60ed9db8c54afe42090d7ab08b631bf0c6c b6a3819cdcd2ebda04207f36cfb009ec36f628dc M	src
bisect run success
```
See: https://github.com/open62541/open62541/commit/02734c310dc4f29a45879eeae1e1b7f63d9c9c12

If you now run the readspeed test on master with:
```
git checkout master
git revert 02734c310dc4f29a45879eeae1e1b7f63d9c9c12
./check_speed.sh
```
It results in a much better readspeed.

So it is almost sure that this commit reduced the speed.

My guess is that the following line is slowing it down significantly:
https://github.com/open62541/open62541/commit/02734c310dc4f29a45879eeae1e1b7f63d9c9c12#diff-298d46de5f9649f904115245ac489be5R103

I'm not sure though...

**Comment by jpfr** [comment]
Wow! That script is pure gold.

Yes. It looks you found the culprit.
The nodeid is printed even though the log level is below the threshold.

\*ouch\*

**Comment by jpfr** [comment]
Fixed via #2048.
